java/code/src/com/redhat/rhn/manager/distupgrade/DistUpgradeManager.java

```diff
@@ -695,7 +696,9 @@ public static Long scheduleDistUpgrade(User user, Server server,
         // Add product upgrades
         // Note: product upgrades are relevant for SLE 10 only!
         if (targetSet != null) {
-            SUSEProductSet installedProducts = server.getInstalledProductSet();
+            SUSEProductSet installedProducts = server.getInstalledProductSet().orElseThrow(() ->
+                    new FaultException(-1, "listMigrationTargetError", "Server has no Products installed."));
```

Comment:
Yes, I believe as this is the final step of scheduling the dist upgrade action we might not even get here in case there is no products installed. But you would also need to follow a more functional approach here I guess in order to avoid the check, or call `get()` which we also want to avoid.

Reply:
In this case, since we don't need to check if `server.getInstalledProductSet()` is defined, and won't have any logic to handle those situations, I would maybe use a plain `get()`.  How could we handle it better?

Reply:
That could work, but it would be a more disruptive change since `scheduleDistUpgrade` is used in a few different places. Can we keep it as it is, in the scope of this PR, and maybe address this change later of needed?

Reply:
Created new issue to address this: https://github.com/SUSE/spacewalk/issues/5684